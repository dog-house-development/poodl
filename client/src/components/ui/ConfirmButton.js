import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MessageModal from './modal/MessageModal';

const propTypes = {
    onConfirm: PropTypes.func.isRequired,
    message: PropTypes.string.isRequired
};

export class ConfirmButton extends Component {
    constructor(props) {
        super(props);
        this.state = {
            confirmModalOpen: false
        };
    }

    handleClick = () => {
        this.setState({ confirmModalOpen: true });
    };

    handleCancel = () => {
        this.setState({ confirmModalOpen: false });
    };

    handleConfirm = () => {
        this.setState({ confirmModalOpen: false });
        this.props.onConfirm();
    };

    getConfirmModal() {
        if (this.state.confirmModalOpen) {
            return (
                <MessageModal
                    title={this.props.title}
                    message={this.props.message}
                    onSubmit={this.handleConfirm}
                    onCancel={this.handleCancel}
                />
            );
        }
    }

    render() {
        return (
            <button onClick={this.handleClick} className={this.props.className}>
                {this.props.children}
                {this.getConfirmModal()}
            </button>
        );
    }
}

ConfirmButton.propTypes = propTypes;
export default ConfirmButton;
