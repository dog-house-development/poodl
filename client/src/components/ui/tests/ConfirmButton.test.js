import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import ConfirmButton from '../ConfirmButton';

configure({ adapter: new Adapter() });

describe('DeleteButton tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = (_props = {}) => {
        const props = _.assign(
            {},
            {
                onConfirm: () => {},
                message: 'Are you sure you want to delete this?'
            },
            _props
        );
        wrapper = shallow(<ConfirmButton {..._.assign({}, props)} />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render delete button correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
