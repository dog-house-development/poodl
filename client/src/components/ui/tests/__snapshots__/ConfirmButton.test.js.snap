// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton tests render should render delete button correctly 1`] = `
<button
  onClick={[Function]}
/>
`;
